src/LibraryManager/Manifest.cs

```diff
@@ -121,12 +121,295 @@ private static bool IsValidManifestVersion(string version)
+            if (string.IsNullOrEmpty(libraryId))
+            {
+                errors.Add(PredefinedErrors.LibraryIdIsUndefined());
+            }
+
+            if (string.IsNullOrEmpty(destination) && string.IsNullOrEmpty(DefaultDestination))
+            {
+                errors.Add(PredefinedErrors.PathIsUndefined());
+            }
+            else
+            {
+                destination = string.IsNullOrEmpty(destination) ? DefaultDestination : destination;
+            }
+
+            IProvider provider = null;
+            if (string.IsNullOrEmpty(providerId) && string.IsNullOrEmpty(DefaultProvider))
+            {
+                errors.Add(PredefinedErrors.ProviderIsUndefined());
+            }
+            else
+            {
+                providerId = string.IsNullOrEmpty(providerId) ? DefaultProvider : providerId;
+
+                provider = _dependencies.Providers.FirstOrDefault(p => p.Id == providerId);
+                if (provider == null)
+                {
+                    errors.Add(PredefinedErrors.ProviderUnknown(providerId));
+                }
+            }
+
+            if (errors.Any())
+            {
+                result = new LibraryInstallationResult(errors);
+                return result;
+            }
+
+            var desiredState = new LibraryInstallationState()
+            {
+                LibraryId = libraryId,
+                Files = files,
+                ProviderId = providerId,
+                DestinationPath = destination
+            };
+
+            List<ILibraryInstallationState> conflictingLibraries = await CheckLibraryForConflictsAsync(desiredState, cancellationToken);
+
+            if (conflictingLibraries != null)
+            {
+                conflictingLibraries.Remove(desiredState);
+                errors.Add(PredefinedErrors.LibraryCannotBeInstalledDueToConflicts(
+                    desiredState.LibraryId,
+                    conflictingLibraries.Select(l => l.LibraryId).ToList()));
+            }
+
+            if (errors.Any())
+            {
+                result = new LibraryInstallationResult(errors);
+                return result;
+            }
+
+            result = await provider.InstallAsync(desiredState, cancellationToken);
+
+            // Remove destination and provider if they match the defaults;
+            if (desiredState.DestinationPath == DefaultDestination)
+            {
+                desiredState.DestinationPath = null;
+            }
+
+            if (desiredState.ProviderId == DefaultProvider)
+            {
+                desiredState.ProviderId = null;
+            }
+
+            if (result.Success)
+            {
+                _libraries.Add(desiredState);
+            }
+
+            return result;
+        }
+
+        private async Task<List<ILibraryInstallationState>> CheckLibraryForConflictsAsync(ILibraryInstallationState desiredState, CancellationToken cancellationToken)
```

Comment:
This check could be called only at RestoreLibraryAsync , if you replace your Install method by RestoreLibraryAsync .